modules/tensor_mechanics/include/materials/ComputeDamageStress.h

```diff
@@ -0,0 +1,39 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifndef COMPUTEDAMAGESTRESS_H
+#define COMPUTEDAMAGESTRESS_H
+
+#include "ComputeFiniteStrainElasticStress.h"
+
+class ComputeDamageStress;
+class DamageBase;
+
+template <>
+InputParameters validParams<ComputeDamageStress>();
+
+/**
+ * ComputeDamageStress computes the stress for a damaged elasticity material. This
+ * model must be used in conjunction with a damage model (derived from DamageBase)
+ */
+class ComputeDamageStress : public ComputeFiniteStrainElasticStress
+{
+public:
+  ComputeDamageStress(const InputParameters & parameters);
+
+  void initialSetup() override;
+
+protected:
+  virtual void computeQpStress() override;
+
+  /// Pointer to the damage model
+  DamageBase * _damage_model;
```

Comment:
I am not sure what is the current convention on the use of raw pointers in MOOSE? If I remember correctly, I was asked to use ``std::shared_ptr`` instead for the linear viscoelasticity classes.

Reply:
I think that raw pointers are OK as long as we're not doing memory allocation (using `new`). In cases like that, we should be using shared pointers to avoid memory leaks.

Reply:
The convention is to consider ownership. If you are passing or sharing ownership of allocated memory, you should be using  a smart pointer. If you are using a pointer for sharing data, a raw pointer is just fine. I think that's a reasonable way to operate in general (not just for MOOSE).

Reply:
In this case, there's no transfer or sharing of ownership, so a raw pointer is fine.

Reply:
@permcody @bwspenc Thanks for the clarification!